worker/jobTypes/productionDeployJob.js

```diff
@@ -0,0 +1,190 @@
+    !currentJob.payload ||
+    !currentJob.payload.repoName ||
+    !currentJob.payload.repoOwner ||
+    !currentJob.payload.branchName
+  ) {
+    workerUtils.logInMongo(
+      currentJob,
+      `${"    (sanitize)".padEnd(15)}failed due to insufficient job definition`
+    );
+    throw invalidJobDef;
+  }
+
+  if (
+    safeString(currentJob.payload.repoName) &&
+    safeString(currentJob.payload.repoOwner) &&
+    safeString(currentJob.payload.branchName)
+  ) {
+    return true;
+  }
+  throw invalidJobDef;
+}
+  // Function that rejects function (promise) after (seconds) seconds with error (errMsg)
+  function promiseTimeoutS(seconds, promise, errMsg) {
+    // Create a promise that rejects in <seconds> seconds
+    const timeout = new Promise((resolve, reject) => {
+      const id = setTimeout(() => {
+        clearTimeout(id);
+        reject(new Error(`${errMsg} --> Timed out in ${seconds} seconds.`));
+      }, 1000 * seconds);
+    });
+
+    // Returns a race between our timeout and the passed in promise
+    return Promise.race([promise, timeout]);
+  }
+async function startGithubBuild(job, logger) {
+  const buildOutput = await workerUtils.promiseTimeoutS(
+    buildTimeout,
+    job.buildRepo(logger),
+    "Timed out on build"
+  );
+  // checkout output of build
+  if (buildOutput && buildOutput.status === "success") {
+    // only post entire build output to slack if there are warnings
+    const buildOutputToSlack = buildOutput.stdout + "\n\n" + buildOutput.stderr;
+    if (buildOutputToSlack.indexOf("WARNING:") !== -1) {
+      await logger.sendSlackMsg(buildOutputToSlack);
+    }
+
+    return new Promise(function(resolve, reject) {
+      resolve(true);
+    });
+  }
+}
+
+async function pushToProduction(publisher, logger) {
+  const prodOutput = await workerUtils.promiseTimeoutS(
+    buildTimeout,
+    publisher.pushToProduction(logger),
+    "Timed out on push to production"
+  );
+  // checkout output of build
+  if (prodOutput && prodOutput.status === "success") {
+    await logger.sendSlackMsg(prodOutput.stdout);
+
+    return new Promise(function(resolve, reject) {
+      resolve(true);
+    });
+  }
+}
+
+async function runGithubProdPush(currentJob) {
+  const ispublishable = await verifyBranchConfiguredForPublish(currentJob);
+  const userIsEntitled = await verifyUserEntitlements(currentJob);
+
+  if (ispublishable === false) {
+    workerUtils.logInMongo(
+      currentJob,
+      `${"(BUILD)".padEnd(15)} You are trying to 
+    
+    to production a branch that is not configured for publishing`
+    );
+    throw new Error("entitlement failed");
+    process.exit;
+  }
+  if (userIsEntitled === false) {
+    workerUtils.logInMongo(
+      currentJob,
+      `${"(BUILD)".padEnd(
+        15
+      )} failed, you are not entitled to build or deploy (${currentJob.payload.repoOwner}/${currentJob.payload.repoName}) for master branch`
+    );
+    throw new Error("entitlement failed");
+    process.exit;
```

Comment:
same here -- I don't think we want to run process.exit here.

Reply:
removed this line